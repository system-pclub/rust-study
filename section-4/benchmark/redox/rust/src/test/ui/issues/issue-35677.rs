use std::collections::HashSet;

fn is_subset<T>(this: &HashSet<T>, other: &HashSet<T>) -> bool {
    this.is_subset(other)
    //~^ ERROR no method named
}

fn main() {}
