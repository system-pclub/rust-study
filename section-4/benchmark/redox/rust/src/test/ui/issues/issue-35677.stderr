error[E0599]: no method named `is_subset` found for type `&std::collections::HashSet<T>` in the current scope
  --> $DIR/issue-35677.rs:4:10
   |
LL |     this.is_subset(other)
   |          ^^^^^^^^^ method not found in `&std::collections::HashSet<T>`
   |
   = note: the method `is_subset` exists but the following trait bounds were not satisfied:
           `T : std::cmp::Eq`
           `T : std::hash::Hash`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
